envconfig.go

```diff
@@ -38,6 +38,14 @@ func Process(prefix string, spec interface{}) error {
 	for i := 0; i < s.NumField(); i++ {
 		f := s.Field(i)
 		if f.CanSet() {
+			if typeOfSpec.Field(i).Anonymous && f.Kind() == reflect.Struct {
+				embeddedPtr := reflect.New(f.Type()).Interface()
```

Comment:
I don't think you need to `New` anything here per: https://github.com/kelseyhightower/envconfig/pull/28/files#diff-c82d24602ac344e5cacfa4a049e5a66eR43